frostsnap_coordinator/src/verify_address.rs

```diff
@@ -0,0 +1,131 @@
+use std::{borrow::BorrowMut, collections::BTreeSet};
+
+use frostsnap_comms::{CoordinatorSendBody, CoordinatorSendMessage, Destination};
+use frostsnap_core::{
+    message::{CoordinatorSend, CoordinatorToDeviceMessage},
+    DeviceId, KeyId,
+};
+use tracing::{event, Level};
+
+use crate::{Completion, Sink, UiProtocol, UiToStorageMessage};
+
+#[derive(Clone, Debug, Default)]
+pub struct VerifyAddressProtocolState {
+    pub target_devices: Vec<DeviceId>,  // not a set for frb compat
+    pub sent_to_devices: Vec<DeviceId>, // not a set for frb compat
+}
+
+pub struct VerifyAddressProtocol {
+    state: VerifyAddressProtocolState,
+    key_id: KeyId,
+    derivation_index: u32,
+    is_complete: Option<Completion>,
+    need_to_send_to: BTreeSet<DeviceId>,
+    sink: Box<dyn Sink<VerifyAddressProtocolState>>,
+}
+
+impl VerifyAddressProtocol {
+    pub fn new(
+        verify_address_messages: &mut Vec<CoordinatorSend>,
```

Comment:
The modern way to do this is to pass a `&FrostCoordinator` in here and call `.verify_address` on it directly. To improve on this it would be more ergonomic to return some kind of `VerifyAddress` which impls `Into<Vec<CoordinatorSend>>` or something like that. This way yo can have an easy time getting hold of the details for this but can also turn it into a `CoordiantorSend` form. This improvement could also be done to `start_sign` which looks where you got this approach from.

Reply:
Instead of passing in the coordinator I decided to just return a `VerifyAddress` from `coordinator.verify_address(...)`, which implements `IntoIterator` since we're always calling `into_iter` anyway.

I was also able to make `coordinator.verify_address` just take `&self`, not mutating anything.

Check it out and lmk what you think.